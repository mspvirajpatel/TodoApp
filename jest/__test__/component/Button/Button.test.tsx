/* eslint-disable jest/valid-expect */
import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import BaseButton from '../../../../app/component/Button/index';

describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = render(<BaseButton label='test label' />);
      expect(component).toMatchSnapshot();
    });
  });
});

describe('Interaction', () => {
  describe('onPressHandler', () => {
    test('clicking the button toggles an answer on/off', () => {
      const onClick = jest.fn();
      const component = render(<BaseButton onClick={onClick} />);
      const button = component.getByTestId('button');
      fireEvent.press(button);
    });
  });
});
// describe('Button Component', () => {
//   test('TextView Snapshot', () => {
//     const wrapper = shallow(<BaseView />);
//     expect(wrapper.debug()).toMatchSnapshot();
//   });

//   // test('baseView Match Snapshot', () => {
//   //   const wrapper = shallow(<BaseView />);
//   //   const instance = wrapper.instance();
//   //   instance.data();
//   //   // expect(wrapper.exists('baseView')).toEqual(true);
//   //   // expect(wrapper.find('baseView')).exists();
//   // });
// });
