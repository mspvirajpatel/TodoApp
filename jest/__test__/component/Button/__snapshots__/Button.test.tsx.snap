// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "stretch",
      "backgroundColor": "#38ba7d",
      "borderBottomColor": "#1e6343",
      "borderBottomWidth": 6,
      "borderWidth": 1,
      "height": 45,
      "justifyContent": "center",
      "marginLeft": 15,
      "marginRight": 15,
      "opacity": 1,
    }
  }
  testID="button"
>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
    testID="text"
  >
    test label
  </Text>
</View>
`;
